src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -122,31 +126,65 @@ inline void JavaThread::clear_obj_deopt_flag() {
   clear_suspend_flag(_obj_deopt);
 }
 
-inline bool JavaThread::clear_async_exception_condition() {
-  bool ret = has_async_exception_condition();
-  if (ret) {
-    clear_suspend_flag(_has_async_exception);
+class AsyncExceptionHandshake : public AsyncHandshakeClosure {
```

Comment:
Should this class still have closure in its name? I find it hard to understand the difference between a XHandshake and a XClosure.